src/Control/Carrier/Telemetry/Utils.hs

```diff
@@ -93,12 +97,40 @@ getCurrentCliEnvironment =
     else CliProductionEnvironment
 
 getSystemInfo :: IO SystemInfo
-getSystemInfo =
+getSystemInfo = do
   SystemInfo
     Info.os
     Info.arch
     <$> Conc.getNumCapabilities
     <*> Conc.getNumProcessors
+    -- Info.os only collects OS type, but for linux there isn't any info about distribution.
+    <*> (if Info.os == "linux" then readOsRelease else pure Nothing)
+    <*> (if Info.os /= "mingw32" then readUname else pure Nothing)
+    <*> if Info.os == "linux" then Just <$> findLddVersion else pure Nothing
+  where
+    -- read more about os-release: https://www.commandlinux.com/man-page/man5/os-release.5.html
+    readOsRelease :: IO (Maybe String)
+    readOsRelease = do
+      releaseInfo <- fmap osRelease <$> parseOsRelease
+      pure $ (pretty_name <$> releaseInfo) <|> (name <$> releaseInfo)
```

Comment:
`parseOsRelease` should catch any `IOException` and return `Nothing`. I think I should refactor it anyways to do my own exception handling so we get a real error message rather than just `Nothing` though.

Reply:
Oh, in that case I think it's fine unless you just have extra time to spend on it.

Reply:
No, this was good. It made me realize that I needed to be a bit more careful in the other parts where I call `readProcess`.

Reply:
Ok, I have incorporated these changes. If you approve/think it's fine I will merge.